ale_python_interface/ale_c_wrapper.cpp

```diff
@@ -1,3 +1,20 @@
-//all code is currently in the .h file
 #include "ale_c_wrapper.h"
 
+#include <cstring>
+#include <string>
+
+void encodeState(ALEState *state, char *buf) {
+	std::string str = state->serialize();
+
+	memcpy(buf, str.data(), str.length());
```

Comment:
What's the rationale for using char *buf instead of std::vector, or simply std::string? In particular, this current form is unsafe since buf may not be of sufficient size.